Modules/getpath.py

```diff
@@ -594,6 +594,12 @@ def search_up(prefix, *landmarks, test=isfile):
         prefix = abspath('')
         warn('Could not find platform independent libraries <prefix>')
 
+    # First try to detect exec_prefix by looking alongside our runtime library, if known
+    if library and not exec_prefix:
+        library_dir = dirname(library)
+        if PLATSTDLIB_LANDMARK:
+            if isdir(joinpath(library_dir, PLATSTDLIB_LANDMARK)):
```

Comment:
My intent was to mimic the blocks that detect prefix -- the block above corresponds to the block for prefix at 553 (the leading comment is nearly identical) and the block for exec_prefix following it corresponds to the block for prefix at 583. I can see how it might be better to rearrange some of the logic and possibly to create a function that's called for each, but the current patch minimizes the number of changes.

Reply:
I can see what the intent was, but now it's harder to follow because there are two apparently independent blocks for calculating `exec_prefix` with no obvious reason. So better to merge it into a single block.